src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -571,26 +571,58 @@ private async Task SearchWithFilter(
 
                 if (_exportJobRecord.ExportType == ExportJobType.Patient || _exportJobRecord.ExportType == ExportJobType.Group)
                 {
-                    uint resultIndex = 0;
-                    foreach (SearchResultEntry result in searchResult.Results)
+                    var searchResultEntries = searchResult?.Results?.ToList();
+                    if (searchResultEntries?.Any() ?? false)
                     {
-                        // If a job is resumed in the middle of processing patient compartment resources it will skip patients it has already exported compartment information for.
-                        // This assumes the order of the search results is the same every time the same search is performed.
-                        if (progress.SubSearch != null && result.Resource.ResourceId != progress.SubSearch.TriggeringResourceId)
+                        var startSurrogateId = sharedQueryParametersList
+                            .Where(x => string.Equals(x.Item1, KnownQueryParameterNames.StartSurrogateId, StringComparison.OrdinalIgnoreCase))
+                            .Select(x => x.Item2)
```

Comment:
I don't think the start and end ids should be used for the compartment search.
I don't know how they are being used, but either they are limiting results based on the surrogate ids which would lead to desired data not being exported or they are being ignored and there is no need to add them.